activesupport/test/core_ext/hash_ext_test.rb

```diff
@@ -1762,4 +1762,11 @@ def test_expansion_count_is_limited
       Hash.from_xml(attack_xml)
     end
   end
+
+  def test_from_xml_doesnt_taint_strings
+    string_object = "<?xml version=\"1.0\"?><a>Cat</a>".freeze
+    string_clone = string_object.clone
+    assert_equal({"a"=>"Cat"}, Hash.from_xml(string_clone))
+    assert_equal string_object, string_clone
+  end
```

Comment:
Pretty much, yeah -- though I'd just put the string literal in `assert_equal` a second time rather than calling `.clone`